src/integTest/java/ai/deepcode/javaclient/DeepCodeRestApiTest.java

```diff
@@ -377,47 +328,60 @@ private EmptyResponse doUploadFile(
+    return DeepCodeRestApi.extendBundle(
+        loggedToken, createBundleResponse.getBundleHash(), ebr);
   }
 
   @Test
-  public void _090_getAnalysis() {
+  public void _090_getAnalysis() throws InterruptedException {
     System.out.println("\n--------------Get Analysis----------------\n");
     assertNotNull(
         "`bundleId` should be initialized at `_030_createBundle_from_source()`", bundleId);
     final String deepcodedFilePath =
-        createFileHashRequest(null).getFiles().keySet().stream().findFirst().orElseThrow();
+        createFileHashRequest(null).keySet().stream().findFirst().orElseThrow();
     final List<String> analysedFiles = Collections.singletonList(deepcodedFilePath);
-    assertAndPrintGetAnalysisResponse(
-        DeepCodeRestApi.getAnalysis(loggedToken, bundleId, null, false, analysedFiles));
-    System.out.println("\n---- With `Linters` param:\n");
-    assertAndPrintGetAnalysisResponse(
-        DeepCodeRestApi.getAnalysis(loggedToken, bundleId, null, true, analysedFiles));
+    GetAnalysisResponse response = doAnalysisAndWait(analysedFiles, null);
+    assertAndPrintGetAnalysisResponse(response);
+    boolean resultsEmpty = response.getSuggestions() == null || response.getSuggestions().isEmpty();
+    assertFalse("Analysis results must not be empty", resultsEmpty);
     System.out.println("\n---- With `severity=2` param:\n");
+    response = doAnalysisAndWait(analysedFiles, 2);
     assertAndPrintGetAnalysisResponse(
-        DeepCodeRestApi.getAnalysis(loggedToken, bundleId, 2, false, analysedFiles));
+        response);
+  }
+
+  @NotNull
+  private GetAnalysisResponse doAnalysisAndWait(List<String> analysedFiles, Integer severity) throws InterruptedException {
+    GetAnalysisResponse response = null;
+    for (int i = 0; i < 120; i++) {
+      response = DeepCodeRestApi.getAnalysis(loggedToken, bundleId, severity, analysedFiles);
+      if (response.getStatus().equals("COMPLETE")) break;
+      Thread.sleep(1000);
+    }
+    return response;
   }
 
   private void assertAndPrintGetAnalysisResponse(GetAnalysisResponse response) {
     assertNotNull(response);
     System.out.printf(
         "Get Analysis call for test file: \n-----------\n %1$s \n-----------\nreturns Status code: %2$s \n%3$s\n",
         testFileContent, response.getStatusCode(), response);
-    //    assertEquals("DONE", response.getStatus());
+    assertEquals("COMPLETE", response.getStatus());
     assertEquals("Get Analysis request not succeed", 200, response.getStatusCode());
   }
 
-  @Test
-  public void setBaseUrl_shouldUseEmptyTrustManager_whenDisableSslVerificationIsTrue() {
-    DeepCodeRestApi.setBaseUrl(baseUrl, true);
-
-    EmptyResponse emptyResponse = DeepCodeRestApi.checkSession(loggedToken);
-
-    assertThat(emptyResponse, notNullValue());
-    assertThat(emptyResponse.getStatusCode(), equalTo(200));
-  }
+//  This test would need a mock of SSLSocketFactory to verify the x509 manager. Commenting the test for now
```

Comment:
What is the need to leave it commented? Let's either document the reason better or remove it :)